File: 2026-09/src/day_08/prms_120878/Solution.py
import math


def solution(a, b):
    q = b // math.gcd(a, b)
    while q % 2 == 0:
        q //= 2
    while q % 5 == 0:
        q //= 5

    return 1 if q == 1 else 2

File: 2026-09/src/day_08/prms_120878/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(int a, int b) {
    int q = b / gcd(a, b);
    while (q % 2 == 0) q /= 2;
    while (q % 5 == 0) q /= 5;

    return q == 1 ? 1 : 2;
}
